src/dmd/dmodule.d

```diff
@@ -66,9 +66,11 @@ private const(char)[] lookForSourceFile(const(char)[] filename)
     const sdi = FileName.forceExt(filename, global.hdr_ext.toDString());
     if (FileName.exists(sdi) == 1)
         return sdi;
+    scope(exit) FileName.free(sdi.ptr);
```

Comment:
the `scope(exit)` is not necessary here

Reply:
Both of these `scope(exit)` are necessary, because `sdi` and `sd` may be used [in the `for`-loop below](https://github.com/dlang/dmd/blob/a032e3cfeb53add32fbb60e6652b1d3515d88e40/src/dmd/dmodule.d#L96-L101).